comp/core/tagger/impl-remote/remote.go

```diff
@@ -250,6 +256,68 @@ func (t *remoteTagger) LegacyTag(entity string, cardinality types.TagCardinality
 	return t.Tag(entityID, cardinality)
 }
 
+// GenerateContainerIDFromExternalData returns a container ID for the given external data.
+func (t *remoteTagger) GenerateContainerIDFromExternalData(externalData origindetection.ExternalData) (string, error) {
+	expBackoff := backoff.NewExponentialBackOff()
+	expBackoff.InitialInterval = 500 * time.Millisecond
+	expBackoff.MaxInterval = 1 * time.Second
+	expBackoff.MaxElapsedTime = 15 * time.Second
+
+	var containerID string
+
+	err := backoff.Retry(func() error {
+		select {
+		case <-t.telemetryTicker.C:
+			t.store.collectTelemetry()
+		case <-t.ctx.Done():
+			return &backoff.PermanentError{Err: errTaggerFailedGenerateContainerIDFromExternalData}
+		default:
+		}
+
+		t.telemetryStore.ExternalDataRequests.Inc()
+
+		// Fetch the auth token
+		token, err := t.options.TokenFetcher()
+		if err != nil {
+			_ = t.log.Errorf("unable to fetch auth token, will possibly retry: %s", err)
+			return err
+		}
+
+		// Create the context with the auth token
+		t.queryCtx, t.queryCancel = context.WithCancel(
+			metadata.NewOutgoingContext(t.ctx, metadata.MD{
+				"authorization": []string{fmt.Sprintf("Bearer %s", token)},
+			}),
+		)
+
+		// Call the GRPC method to get the container ID from the external data
+		containerIDResponse, err := t.client.TaggerGenerateContainerIDFromExternalData(t.queryCtx, &pb.GenerateContainerIDFromExternalDataRequest{
+			Init:          externalData.Init,
+			ContainerName: externalData.ContainerName,
+			PodUID:        externalData.PodUID,
+		})
+		if err != nil {
+			_ = t.log.Errorf("unable to generate container ID from external data, will retry: %s", err)
+			return err
+		}
+
+		if containerIDResponse == nil {
+			_ = t.log.Warnf("unable to generate container ID from external data, will retry: %s", err)
+			return errors.New("containerIDResponse is nil")
+		}
+		containerID = containerIDResponse.ContainerID
+
+		t.telemetryStore.ExternalDataRequests.Inc()
```

Comment:
I think this should increase "Success" instead of "Requests"

Reply:
Fixed 👍